prawcore/auth.py

```diff
@@ -167,6 +167,34 @@ def revoke(self, only_access=False):
             self.refresh_token = None
 
 
+class DeviceIDAuthorizer(Authorizer):
+    """Manages app-only OAuth2 for 'installed' applications.
+
+    While the '*' scope will be available, some endpoints simply will not work
+    due to the lack of an associated reddit account.
+    """
+
+    def __init__(self, authenticator, device_id):
+        """Represents an app-only OAuth2 authorization for 'installed' apps.
+
+        :param authenticator: An instance of :class:`Authenticator`.
+        :param device_id: A unique ID (20-30 character ASCII string)
+            For more information about this parameter, see:
+            https://github.com/reddit/reddit/wiki/OAuth2#application-only-oauth
+        """
+        if authenticator.client_secret != '':
+            raise InvalidInvocation(
+                'confindential client cannot instantiate this class')
```

Comment:
spelling: confidential

I wonder if rather than an empty string, `client_secret` shouldn't be set at all. I'll think about that as I review the rest.